From d077f0ba2c5b40930f95d59d03d0bb4f68a64245 Mon Sep 17 00:00:00 2001
From: Madalin Bucur <madalin.bucur@freescale.com>
Date: Fri, 21 Feb 2014 17:04:08 +0200
Subject: [PATCH 557/987] dpaa_eth: move declarations to the proper location

[Original patch taken from QorIQ-SDK-V1.6-SOURCE-20140619-yocto.iso]

Move dpa_alloc_pcd_fqids and dpa_free_pcd_fqids to dpaa_eth_common.c.

Signed-off-by: Madalin Bucur <madalin.bucur@freescale.com>
Change-Id: I6322cdc377e2b8061680fc39129d567726ec4c29
Reviewed-on: http://git.am.freescale.net:8181/9026
Reviewed-by: Cristian-Constantin Sovaiala <Cristian.Sovaiala@freescale.com>
(cherry picked from commit fe1d9e3e097efe1c330f4209b7cca7f75049a6a8)
Reviewed-on: http://git.am.freescale.net:8181/9191
---
 drivers/net/ethernet/freescale/dpa/dpaa_eth.c        |  8 --------
 drivers/net/ethernet/freescale/dpa/dpaa_eth_common.c | 14 ++++++++++++++
 2 files changed, 14 insertions(+), 8 deletions(-)

diff --git a/drivers/net/ethernet/freescale/dpa/dpaa_eth.c b/drivers/net/ethernet/freescale/dpa/dpaa_eth.c
index b6435d3..ab7658b 100644
--- a/drivers/net/ethernet/freescale/dpa/dpaa_eth.c
+++ b/drivers/net/ethernet/freescale/dpa/dpaa_eth.c
@@ -102,14 +102,6 @@ static const char rtx[][3] = {
 	[TX] = "TX"
 };
 
-#if defined(CONFIG_FSL_FMAN_TEST)
-/* Defined as weak, to be implemented by fman pcd tester. */
-int dpa_alloc_pcd_fqids(struct device *, uint32_t, uint8_t, uint32_t *)
-__attribute__((weak));
-
-int dpa_free_pcd_fqids(struct device *, uint32_t) __attribute__((weak));
-#endif /* CONFIG_FSL_FMAN_TEST */
-
 /* BM */
 
 #define DPAA_ETH_MAX_PAD (L1_CACHE_BYTES * 8)
diff --git a/drivers/net/ethernet/freescale/dpa/dpaa_eth_common.c b/drivers/net/ethernet/freescale/dpa/dpaa_eth_common.c
index 688f8b8..a62bba1 100644
--- a/drivers/net/ethernet/freescale/dpa/dpaa_eth_common.c
+++ b/drivers/net/ethernet/freescale/dpa/dpaa_eth_common.c
@@ -1414,6 +1414,20 @@ dpaa_eth_init_rx_port(struct fm_port *port, struct dpa_bp *bp, size_t count,
 			   buf_layout, frag_enabled);
 }
 
+#if defined(CONFIG_FSL_FMAN_TEST)
+/* Defined as weak, to be implemented by fman pcd tester. */
+int dpa_alloc_pcd_fqids(struct device *, uint32_t, uint8_t, uint32_t *)
+__attribute__((weak));
+
+int dpa_free_pcd_fqids(struct device *, uint32_t) __attribute__((weak));
+#else
+int dpa_alloc_pcd_fqids(struct device *, uint32_t, uint8_t, uint32_t *);
+
+int dpa_free_pcd_fqids(struct device *, uint32_t);
+
+#endif /* CONFIG_FSL_FMAN_TEST */
+
+
 int dpa_alloc_pcd_fqids(struct device *dev, uint32_t num,
 				uint8_t alignment, uint32_t *base_fqid)
 {
-- 
1.9.1

